cmd/mistryd/server.go

```diff
@@ -118,28 +126,82 @@ func (s *Server) HandleNewJob(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// send the work item to the worker pool
+	workItem := WorkItem{j, jr, make(chan WorkResult, 1)}
```

Comment:
does this have to be buffered?

Reply:
yes, the workers need to put the result there and move on to the next work item instead of waiting for the reader of this channel to receive the result, which may happen immediately, later, or never